### PR TITLE
chore(logs): add output format option to `karmor logs` command

### PR DESCRIPTION
**Pull request by navin772**
Fixes: #439

Introduced new flags for pretty and normal printing:

`karmor logs --output json`
`karmor logs --output pretty-json`
`karmor logs --output text`

Shorthand -> `-o`

The `--json` flag is still present:
`karmor logs --json`

### Output:
```bash
$ ./karmor logs -o pretty-json

local port to be used for port forwarding kubearmor-relay-8464877449-sp5mt: 32816 
Created a gRPC client (localhost:32816)
Checked the liveness of the gRPC server
Started to watch alerts

{
  "Timestamp": 1719459063,
  "UpdatedTime": "2024-06-27T03:31:03.302163Z",
  "ClusterName": "default",
  "HostName": "navin",
  "HostPPID": 8384,
  "HostPID": 14768,
  "PPID": 8384,
  "PID": 14768,
  "UID": 1000,
  "ParentProcessName": "/usr/bin/bash",
  "ProcessName": "/usr/bin/date",
  "PolicyName": "hsp-kubearmor-dev-proc-path-block-fromsource",
  "Severity": "5",
  "Type": "MatchedHostPolicy",
  "Source": "/usr/bin/bash",
  "Operation": "Process",
  "Resource": "/usr/bin/date",
  "Data": "lsm=SECURITY_BPRM_CHECK",
  "Enforcer": "BPFLSM",
  "Action": "Block",
  "Result": "Permission denied",
  "Cwd": "/"
}
```

### CODE REVIEW

File: log/logClient.go [hunk 1]
```diff
@@ -407,6 +408,19 @@ func WatchTelemetryHelper(arr []byte, t string, o Options) {
 	}
 
 	if o.JSON {
+		str = fmt.Sprintf("%s\n", string(arr))
+	} else if o.Output == "pretty-json" {
```
Comment (daemon1024) [inline log/logClient.go:412]:
```suggestion
	if o.JSON || o.Output == "json" {
		str = fmt.Sprintf("%s\n", string(arr))
	} else if o.Output == "pretty-json" {
```

**Review by daemon1024 — CHANGES_REQUESTED** [review]

File: log/logClient.go [hunk 2]
```diff
@@ -406,7 +407,20 @@ func WatchTelemetryHelper(arr []byte, t string, o Options) {
...
+		var prettyJSON bytes.Buffer
+		err = json.Indent(&prettyJSON, arr, "", "  ")
+
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to prettify JSON (%s)\n", err.Error())
+		}
+		str = fmt.Sprintf("%s\n", prettyJSON.String())
+
+	} else if o.Output == "json" {
+
```
Comment (daemon1024) [inline log/logClient.go:423]: This is redundant now, since we added it in the earlier check

**Review by daemon1024 — CHANGES_REQUESTED** [review]

File: log/logClient.go (hunk 2 above)
Comment (navin772) [reply log/logClient.go]: Thanks for pointing out, forgot that!

**Review by daemon1024 — APPROVED** [review]

**Comment by daemon1024** [comment]
Can we squash the commits please? Thanks

**Comment by DelusionalOptimist** [comment]
@navin772 please rebase with upstream main once to fix the failing tests.

**Review by daemon1024 — APPROVED** [review]

**Comment by daemon1024** [comment]
Please squash the commits 🙌 Rest LGTM

**Comment by daemon1024** [comment]
Please squash the commits 🙌 Rest LGTM

**Pull request merged**